apps/api/src/controllers/post.ts

```diff
@@ -0,0 +1,109 @@
+import Post from '../models/Post';
+import Comment from '../models/Comment';
+
+//Get posts list
+const getPosts = async (req, res) => {
+  try {
+    const posts = await Post.find();
+    res.status(200).json(posts);
+  } catch (error) {
+    const { message } = error;
+    res.status(500).json({ message });
+  }
+};
+
+//Get post by category
+const getPostsByCategory = async (req, res) => {
+  try {
+    const { category } = req.params;
+    const post = await Post.find({ category }).populate('category');
+    if (!post) {
+      return res.status(404).json({ message: 'Posts not found' });
+    }
+    res.status(200).json(post);
+  } catch (error) {
+    const { message } = error;
+    res.status(500).json({ message });
+  }
+};
+
+//Get post by id
+const getPostById = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const post = await Post.findById(id).populate('category');
```

Comment:
FYI, since we are going to display the comments in the detail page, you need to populate `comments` as well.